src/pages/home/report/ReportActionsList.js

```diff
@@ -185,25 +195,41 @@ function ReportActionsList({
             return;
         }
 
+        cacheUnreadMarkers.delete(report.reportID);
         reportActionSize.current = sortedReportActions.length;
         setCurrentUnreadMarker(null);
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [sortedReportActions.length, report.reportID]);
 
     useEffect(() => {
-        const didManuallyMarkReportAsUnread = report.lastReadTime < DateUtils.getDBTime() && ReportUtils.isUnread(report);
-        if (didManuallyMarkReportAsUnread) {
-            // Clearing the current unread marker so that it can be recalculated
-            setCurrentUnreadMarker(null);
-            setMessageManuallyMarkedUnread(new Date().getTime());
+        if (!userActiveSince.current || report.reportID !== prevReportID.current) {
             return;
         }
-
+        if (!messageManuallyMarkedUnread && lastReadRef.current && lastReadRef.current < report.lastReadTime) {
```

Comment:
where is messageManuallyMarkedUnread initialized?

Reply:
`const [messageManuallyMarkedUnread, setMessageManuallyMarkedUnread] = useState(0);` Line 159

Reply:
Thanks! I swear, I searched